colcon_installed_package_information/package_augmentation/python.py

```diff
@@ -42,18 +42,21 @@ def augment_packages(  # noqa: D102
                 name=desc.name)
             try:
                 dist = next(iter(dists))
-            except IndexError:
+            except StopIteration:
                 continue
 
-            version = dist.version
-            if version and not desc.metadata.get('version'):
-                desc.metadata['version'] = version
+            if not desc.metadata.get('version'):
```

Comment:
This is a nit, but since `get` returns None by default:

```suggestion
            if desc.metadata.get('version') is not None:
```

Reply:
Thanks for taking a look. From what I can tell, the behavior of an empty version string is largely undefined. In at least `colcon-package-information`, it gets treated the same as if it were undefined. Given that, I'd like to keep this as-is in this change. We can revisit it another time if you're interested.